src/nvim/api/buffer.c

```diff
@@ -86,6 +86,12 @@ void buffer_del_line(Buffer buffer, Integer index, Error *err)
   FUNC_ATTR_DEFERRED
 {
   Array array = ARRAY_DICT_INIT;
+  buf_T *buf = find_buffer_by_handle(buffer, err);
+
+  if (index >= buf->b_ml.ml_line_count) {
+    return;
+  }
+
```

Comment:
Actually, thinking ahead to when an Exception is returned when writing to a read-only buffer, it might be better to just pass through the exception to avoid debugging pains when a delete succeeds but is followed by a prevented write. What do you think?